### PR TITLE
generics: Add NullablePtr

### PR DESCRIPTION
**Pull request by xavier-petit**
### What does this PR do?

Could be used instead of type specific version `stringutil.NullablePtr` & `int64.NullablePtr`

Fixes #

### Good PR checklist

- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [ ] Added/updated documentation
- [x] Properly labeled

### CODE REVIEW

**Review by Sypheos — COMMENTED** [review]
Thx !

File: generics/ptr_test.go [hunk 1]
```diff
@@ -16,6 +16,14 @@ func TestPointer(t *testing.T) {
 	)
 }
 
+func TestNullablePtr(t *testing.T) {
+	assert.Equal(t, *NullablePtr[int](-5), -5)
+	assert.Equal(t, NullablePtr[int](0), (*int)(nil))
```
[the remark this answers is not in the document]
Comment (xavier-petit) [reply generics/ptr_test.go:21]: Oh right, thank you!

**Comment by AlexisMontagne** [comment]
LGTM! can you add a `// Deprecated: Use generics.NullablePtr instead` comment over  the methods in `stringutil` and `int64util` so we can get rid of those in the next major version

**Review by xgoffin — APPROVED** [review]
:muscle:

**Review by Sypheos — APPROVED** [review]

**Comment by xavier-petit** [comment]
> LGTM! can you add a `// Deprecated: Use generics.NullablePtr instead` comment over the methods in `stringutil` and `int64util` so we can get rid of those in the next major version

Done! Also added Batch @Sypheos

**Pull request merged**